### PR TITLE
Suggested changes for sections: energy demand & tech change

### PR DESCRIPTION
**Pull request by Jihoon**
Minor edits and improvements

In technological change section, it would be helpful to add remarks on how general diffusion constraints are determined for different technologies or what the values are based on.

### CODE REVIEW

**Comment by Jihoon** [comment]
For 'Energy demand':
- Clarified the key assumption under the quantile regression etc.
- The regression part was extremely difficult to read. I separated out the dependent variables for the regional regression as bullets.
- 'the user' -> 'users'
- Removed the uses of 'several'
- "Several user-defined inputs allow..." <- What are these exactly?
- "Given these input parameters, ..." <- By carefully reading it, I guess these inputs indicate the convergence quantile and income. I added them in the text.
- Table captions indicate the dependent variables (FEI, shares, etc.) as 'parameter', which generates confusion with the 'input parameters' in the text. Changed to 'variable'

**Comment by jkikstra** [comment]
@Jihoon I was assigned to check up on this PR. A few comments:

I very much like the textual changes in the documentation, it is much clearer now.

Things that should be changed:
1. I believe that these changes should first be committed to #20 (https://github.com/iiasa/message_doc/tree/2019-update) rather than directly to the master.

Minor textual points that could be considered:
2. Rather write "quantity" than "variable", to be fully generic in the description of the tables.

3. A space is missing after `*t*` for _t_=0 at the line "Fig. 13 below illustrates the maximum technology growth starting at a level of 1 in year *t*=0 for ... "

File: source/energy/demand.rst [hunk 1]
```diff
@@ -23,14 +23,24 @@ sources for the historical and projected datasets are the following:
...
 
 The total final energy in each region is then calculated by multiplying the extrapolated final energy intensity by the projected GDP (PPP) in each time period. Next, the extrapolated shares are multiplied by the total final energy to identify final energy demand for each of the seven energy service demands used in MESSAGE. Finally, final energy is converted to useful energy in each region by using the average final-to-useful energy efficiencies used in the MESSAGE model for each model region (:ref:`spatial`).
 
 .. _tab-quantssp1:
-.. table:: Convergence quantile and income for each parameter and region for SSP1 (for region descriptions, see: :ref:`spatial`)
+.. table:: Convergence quantile and income for each variable and region for SSP1 (for region descriptions, see: :ref:`spatial`)
```
Comment (jkikstra) [inline source/energy/demand.rst]: perhaps say 'quantity'

Comment (Jihoon) [reply source/energy/demand.rst]: Thanks, @jkikstra. It sounds better.

**Comment by Jihoon** [comment]
replaced with #34

**Pull request closed without merging**